### PR TITLE
fix: replace "owner" label with "space" label

### PR DESCRIPTION
**Pull request by mfrancisc**
Check that new `SpaceLabelKey` is present on NSTeamplateResources

Jira:https://issues.redhat.com/browse/CRT-1776

Paired with: https://github.com/codeready-toolchain/member-operator/pull/438

### CODE REVIEW

**Review by alexeykazakov — APPROVED** [review]

**Comment by mfrancisc** [comment]
I've added more checks after the approval, since the label was not verified in "all" the resources , now it should be.
Only exception are ServiceAccounts , as said in the [comment of the paired member-operator PR](https://github.com/codeready-toolchain/member-operator/pull/438#issuecomment-1546708063), I'm not sure if we want to update those and also check the presence of the label in the future.

File: testsupport/tiers/checks.go [hunk 1]
```diff
@@ -355,6 +362,43 @@ func (a *baseextendedidlingTierChecks) GetClusterObjectChecks() []clusterObjects
...
+func commonToolchainLabelsRoleBindingsCheck() spaceRoleObjectsCheck {
+	return func(t *testing.T, ns *corev1.Namespace, memberAwait *wait.MemberAwaitility, userName string) {
+		roleBindings := &rbacv1.RoleBindingList{}
+		err := memberAwait.Client.List(context.TODO(), roleBindings, providerMatchingLabels, client.InNamespace(ns.Name))
+		require.NoError(t, err)
+		for i := range roleBindings.Items {
+			assertExpectedToolchainLabels(t, &roleBindings.Items[i], userName)
+		}
+	}
+}
```
Comment (MatousJobanek) [inline testsupport/tiers/checks.go:391]: out of curiosity - why do we need this here? I thought that we already verify the labels as part of the checks for the roles and rolebindings... But I maybe missed something

File: testsupport/tiers/checks.go [hunk 2]
```diff
@@ -603,8 +653,9 @@ func userEditRoleBinding(userName string) spaceRoleObjectsCheck {
 		assert.Equal(t, "edit", rb.RoleRef.Name)
 		assert.Equal(t, "ClusterRole", rb.RoleRef.Kind)
 		assert.Equal(t, "rbac.authorization.k8s.io", rb.RoleRef.APIGroup)
-		assert.Equal(t, "codeready-toolchain", rb.ObjectMeta.Labels["toolchain.dev.openshift.com/provider"])
-		assert.Equal(t, owner, rb.ObjectMeta.Labels["toolchain.dev.openshift.com/owner"])
+		assert.Equal(t, toolchainv1alpha1.ProviderLabelValue, rb.ObjectMeta.Labels[toolchainv1alpha1.ProviderLabelKey])
+		assert.Equal(t, owner, rb.ObjectMeta.Labels[toolchainv1alpha1.OwnerLabelKey])
+		assert.Equal(t, owner, rb.ObjectMeta.Labels[toolchainv1alpha1.SpaceLabelKey])
```
Comment (MatousJobanek) [inline testsupport/tiers/checks.go:658]: could these occurrences be replaced by `assertExpectedToolchainLabels` as well?

File: testsupport/tiers/checks.go [hunk 3]
```diff
@@ -355,6 +362,43 @@ func (a *baseextendedidlingTierChecks) GetClusterObjectChecks() []clusterObjects
 		idlers(518400, "dev", "stage"))
 }
 
+func commonToolchainLabelsNamespaceCheck() namespaceObjectsCheck {
+	return func(t *testing.T, ns *corev1.Namespace, _ *wait.MemberAwaitility, userName string) {
+		assertExpectedToolchainLabels(t, ns, userName)
+	}
+}
```
Comment (MatousJobanek) [inline testsupport/tiers/checks.go:369]: why do we need this check when we verify the set of labels as part of the `WaitForNamespace` method?
https://github.com/codeready-toolchain/toolchain-e2e/blob/1222fd9e4a79ca6d7cbfa357e5d980aa97536ff7/testsupport/wait/member.go#L653-L665

File: testsupport/tiers/checks.go [hunk 4]
```diff
@@ -1733,27 +1797,30 @@ func toolchainSaReadRole() namespaceObjectsCheck {
...
 
 func namespaceManagerSA() namespaceObjectsCheck {
-	return func(t *testing.T, ns *corev1.Namespace, memberAwait *wait.MemberAwaitility, _ string) {
-		serviceAccount, err := memberAwait.WaitForServiceAccount(t, ns.Name, toolchainv1alpha1.AdminServiceAccountName)
+	return func(t *testing.T, ns *corev1.Namespace, memberAwait *wait.MemberAwaitility, owner string) {
+		_, err := memberAwait.WaitForServiceAccount(t, ns.Name, toolchainv1alpha1.AdminServiceAccountName)
 		require.NoError(t, err)
-		assert.Equal(t, "codeready-toolchain", serviceAccount.ObjectMeta.Labels["toolchain.dev.openshift.com/provider"])
+		// fixme: decide if we want to check labels also on serviceaccounts
+		//assertExpectedToolchainLabels(t, sa, owner)
```
[the remark this answers is not in the document]
Comment (mfrancisc) [reply testsupport/tiers/checks.go:1769]: Got it! 👍

File: testsupport/tiers/checks.go (hunk 2 above)
Comment (mfrancisc) [reply testsupport/tiers/checks.go]: I've missed those, thanks!

Done in https://github.com/codeready-toolchain/toolchain-e2e/pull/726/commits/42199b69c34bf852ef4019bb74e6286bf62ff756

File: testsupport/tiers/checks.go (hunk 3 above)
Comment (mfrancisc) [reply testsupport/tiers/checks.go]: I've removed this check in https://github.com/codeready-toolchain/toolchain-e2e/commit/fca5ee35c5ec4ed8de9397607a70ab6f1eac5d23 . Let's see how it goes without that check.

 I remember facing some issues with the presence of the label that was updated with some delay by the new version of the operator during the migration tests. But double checking it now, I agree it doesn't seem to be needed ! 👍

**Comment by mfrancisc** [comment]
/retest

infra

**Review by MatousJobanek — APPROVED** [review]

**Comment by mfrancisc** [comment]
@MatousJobanek thanks for approval! But I'm still struggling to understand why is it failing  to find these labels in CI:

```
    checks.go:362: 
        	Error Trace:	/go/src/github.com/codeready-toolchain/toolchain-e2e/test/migration/verify/checks.go:362
        	            				/go/src/github.com/codeready-toolchain/toolchain-e2e/test/migration/verify/checks.go:632
        	            				/go/src/github.com/codeready-toolchain/toolchain-e2e/test/migration/verify/nstemplateset.go:54
        	            				/go/src/github.com/codeready-toolchain/toolchain-e2e/test/migration/verify/asm_amd64.s:1594
        	Error:      	map[string]string{"toolchain.dev.openshift.com/owner":"mig-m2-space", "toolchain.dev.openshift.com/provider":"codeready-toolchain"} does not contain "toolchain.dev.openshift.com/space"
        	Test:       	TestAfterMigration
    checks.go:363: 
        	Error Trace:	/go/src/github.com/codeready-toolchain/toolchain-e2e/test/migration/verify/checks.go:363
        	            				/go/src/github.com/codeready-toolchain/toolchain-e2e/test/migration/verify/checks.go:632
        	            				/go/src/github.com/codeready-toolchain/toolchain-e2e/test/migration/verify/nstemplateset.go:54
        	            				/go/src/github.com/codeready-toolchain/toolchain-e2e/test/migration/verify/asm_amd64.s:1594
        	Error:      	Not equal: 
        	            	expected: "mig-m2-space"
        	            	actual  : ""
        	            	
        	            	Diff:
        	            	--- Expected
        	            	+++ Actual
        	            	@@ -1 +1 @@
        	            	-mig-m2-space
        	            	+
```

I've ran 3 times the tests locally and they always pass. Maybe I'm blind or the PR is not paired correctly 🤔

**Comment by mfrancisc** [comment]
Yeah, that is just my merge message, I've fixed that merge conflict, pushed with that string as commit message.

So I guess:
1. is something "stupid" that I'm missing
2. in CI the test is very quick in doing:
 a) setups, 
 b) deploy new operator
 c) verify 
 and the new operator doesn't have enough time to put those new labels on the resources.
 While locally it works since there is more delay between b) and c) 🤷‍♂️

File: testsupport/tiers/checks.go [hunk 5]
```diff
@@ -1070,16 +1071,23 @@ func clusterObjectsChecks(checkCreator ...clusterObjectsCheckCreator) []clusterO
...
 	return func() clusterObjectsCheck {
 		return func(t *testing.T, memberAwait *wait.MemberAwaitility, userName, tierLabel string) {
+			idlerWaitCriterion := []wait.IdlerWaitCriterion{
+				wait.IdlerHasTier(tierLabel),
+				wait.IdlerHasTimeoutSeconds(timeoutSeconds),
+			}
+			// cast generic labels wait criterion into idler wait criterion
+			for expectedKey, expectedValue := range toolchainLabels(userName) {
+				idlerWaitCriterion = append(idlerWaitCriterion, wait.IdlerHasLabel(expectedKey, expectedValue))
+			}
```
Comment (MatousJobanek) [inline testsupport/tiers/checks.go:1081]: We could maybe start using generics for the wait criteria so we could easily reuse them in all functions. 
but that's for later - not now :wink:

**Review by MatousJobanek — APPROVED** [review]
Looks good :+1: great job :1st_place_medal: :rocket:

File: testsupport/tiers/checks.go (hunk 5 above)
Comment (mfrancisc) [reply testsupport/tiers/checks.go:1081]: Yeah I had the same feeling while writing this code :)

**Pull request merged**